### PR TITLE
glib: fix ABI issues with macros/constants

### PR DESCRIPTION
**Pull request by jf2048**
Part of https://github.com/gtk-rs/gtk-rs-core/issues/64

- `g_macro__has_attribute___noreturn__` and `G_LOG_DOMAIN` are used for C macros, should not be in ffi at all
- `G_LOG_LEVEL_MASK` is considered unsigned by rust but C defines it as signed

### CODE REVIEW

File: GLib-2.0.gir [hunk 1]
```diff
@@ -12890,32 +12890,6 @@ module.</doc>
       <doc xml:space="preserve">Multiplying the base 2 exponent by this number yields the base 10 exponent.</doc>
       <type name="gdouble" c:type="gdouble"/>
     </constant>
-    <constant name="LOG_DOMAIN" value="0" c:type="G_LOG_DOMAIN">
```
Comment (bilelmoussaoui) [inline GLib-2.0.gir:12893]: Instead this can be ignored in the Gir.toml file which will drop it from the ffi tests

File: GLib-2.0.gir [hunk 2]
```diff
@@ -13884,7 +13858,7 @@ levels using g_log_set_handler() and g_log_set_fatal_mask().</doc>
       <member name="level_debug" value="128" c:identifier="G_LOG_LEVEL_DEBUG">
         <doc xml:space="preserve">log level for debug messages, see g_debug()</doc>
       </member>
-      <member name="level_mask" value="-4" c:identifier="G_LOG_LEVEL_MASK">
+      <member name="level_mask" value="4294967292" c:identifier="G_LOG_LEVEL_MASK">
```
Comment (bilelmoussaoui) [inline GLib-2.0.gir:13861]: shouldn't this be fixed upstream?

File: GLib-2.0.gir [hunk 3]
```diff
@@ -45766,9 +45740,6 @@ application domain</doc>
         </parameter>
       </parameters>
     </function-macro>
-    <constant name="macro__has_attribute___noreturn__" value="0" c:type="g_macro__has_attribute___noreturn__">
```
Comment (bilelmoussaoui) [inline GLib-2.0.gir:45769]: Similar to the other constant

File: GLib-2.0.gir (hunk 2 above)
Comment (sdroege) [reply GLib-2.0.gir:13861]: You can add `(type uint)` annotation for this IIRC

Comment (sdroege) [reply GLib-2.0.gir:13861]: And `(value 4294967292)`. That's done in GStreamer in many places. It also confuses the Python bindings if you don't do that, but I guess nobody is using `G_LOG_LEVEL_MASK` from Python.

File: GLib-2.0.gir (hunk 1 above)
Comment (jf2048) [reply GLib-2.0.gir:12893]: I tried that but couldn't get it to work, let me try again

File: GLib-2.0.gir (hunk 2 above)
Comment (jf2048) [reply GLib-2.0.gir:13861]: I think we have this problem with any flags type, we consider them u32 but in C anything in an enum has to be `int`. Maybe I will just have gir cast them to unsigned

File: GLib-2.0.gir (hunk 1 above)
Comment (bilelmoussaoui) [reply GLib-2.0.gir:12893]: we do that in gtk for the version constants for exampel

File: GLib-2.0.gir (hunk 2 above)
Comment (jf2048) [reply GLib-2.0.gir:13861]: Yeah it looks like those annotations won't work on an enum constant, they only work on a `#define`

**Comment by jf2048** [comment]
Closing in favor of more stuff in #1409
